src/consensus/chainvalid.md

```diff
@@ -0,0 +1,116 @@
+# Chain validity
+
+A chain of blocks has to be valid in order to be considered for adoption. In
+particular the mainnet chain _is_ valid.
+
+Validity is defined by induction on the length of the chain as:
+- The first block on the chain fragment is valid,
+- The tail of the chain fragment is valid.
+
+For a block to be considered valid it has to be valid in three directions:
+
+- The envelope of the header must be valid, validity in this dimension is a
```

Comment:
Should we defined what the header envelope is?

Reply:
Oh, we do later on. Maybe we should forward link to that section or explain it earlier.